brite2caom2/storage_name.py

```diff
@@ -0,0 +1,181 @@
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  : 4 $
+#
+# ***********************************************************************
+#
+
+from os.path import basename, dirname, exists, splitext
+from caom2pipe.manage_composable import StorageName
+
+
+__all__ = ['add_entry', 'BriteName']
+
+
+class BriteName(StorageName):
+    """
+    For the decorrelated planes, need to have the content of multiple files to generate the previews, so implement
+    this class to collect the artifacts by plane product ID. This decision is captured in how _source_names and
+    _destination_uris are set.
+    """
+
+    def __init__(self, entry):
+        super().__init__(
+            file_name=basename(entry),
+            source_names=[entry],
+        )
+
+    @property
+    def average_uri(self):
+        return self._get_uri(f'{self._obs_id}.avedb')
+
+    @property
+    def decorrelated_uri(self):
+        return self._get_uri(f'{self._obs_id}.ndatdb')
+
+    @property
+    def has_decorrelated_metadata(self):
+        return self._file_name.endswith('.ndatdb')
+
+    @property
+    def has_undecorrelated_metadata(self):
+        return self._file_name.endswith('.orig')
+
+    @property
+    def has_data(self):
+        return (
+            self._file_name.endswith('.orig')
+            or self._file_name.endswith('.ndatdb')
+            or self._file_name.endswith('.avedb')
+        )
+
+    @property
+    def is_last_to_ingest(self):
+        return self._file_name.endswith('.rlogdb')
+
+    @property
+    def prev(self):
+        result = f'{self._obs_id}_prev.jpg'
+        if self.has_undecorrelated_metadata:
+            result = f'{self._obs_id}_un_prev.jpg'
+        return result
+
+    @property
+    def thumb(self):
+        result = f'{self._obs_id}_prev_256.jpg'
+        if self.has_undecorrelated_metadata:
+            result = f'{self._obs_id}_un_prev_256.jpg'
+        return result
+
+    def is_valid(self):
+        return True
+
+    def rename(self, original_extension, new_extension):
+        """
+        Common code to refer to a different file than the one currently being processed.
+        :return:
+        """
+        new_fqn = self._source_names[0].replace(original_extension, new_extension)
+        new_uri = self._get_uri(basename(new_fqn))
+        return new_fqn, new_uri
+
+    def set_file_id(self):
+        self._file_id = BriteName.remove_extensions(self._file_name)
+
+    def set_product_id(self):
+        self._product_id = 'decorrelated'
+        if self.has_undecorrelated_metadata:
+            self._product_id = 'un-decorrelated'
+
+    @staticmethod
+    def remove_extensions(f_name):
+        return splitext(f_name)[0]
+
+    @staticmethod
+    def archived(file_name):
```

Comment:
I would use `is_archived` like the other similar methods in this class.

Reply:
Changed.